test/e2e/specs/editor/blocks/buttons.spec.js

```diff
@@ -162,6 +162,83 @@ test.describe( 'Buttons', () => {
 		);
 	} );
 
+	test( 'can toggle button link settings', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await editor.insertBlock( { name: 'core/buttons' } );
+		await page.keyboard.type( 'WordPress' );
+		await pageUtils.pressKeys( 'primary+k' );
+		await page.keyboard.type( 'https://www.wordpress.org/' );
+		await page.keyboard.press( 'Enter' );
+
+		// Edit link.
+		await page.getByRole( 'button', { name: 'Edit' } ).click();
+
+		// Open Advanced settings panel.
+		await page
+			.getByRole( 'region', {
+				name: 'Editor content',
+			} )
+			.getByRole( 'button', {
+				name: 'Advanced',
+			} )
+			.click();
+
+		// Navigate to and toggle the "Open in new tab" checkbox.
+		const checkbox = page.getByLabel( 'Open in new tab' );
+		await checkbox.click();
+
+		// Toggle should still have focus and be checked.
+		await expect( checkbox ).toBeChecked();
+		await expect( checkbox ).toBeFocused();
+
+		// Tab back to the Save button and apply the link.
+		await page
+			//TODO: change to a better selector when https://github.com/WordPress/gutenberg/issues/51060 is resolved.
+			.locator( '.block-editor-link-control' )
+			.getByRole( 'button', { name: 'Save' } )
+			.click();
+
+		// Check the content.
+		const content = await editor.getEditedPostContent();
+		expect( content ).toBe(
+			`<!-- wp:buttons -->
+<div class="wp-block-buttons"><!-- wp:button -->
+<div class="wp-block-button"><a class="wp-block-button__link wp-element-button" href="https://www.wordpress.org/" target="_blank" rel="noreferrer noopener">WordPress</a></div>
+<!-- /wp:button --></div>
+<!-- /wp:buttons -->`
+		);
```

Comment:
You could do something like this instead to make the test more readable:

https://github.com/WordPress/gutenberg/blob/b05bee7e2e8eefb2be06c228002e4f9783847747/test/e2e/specs/editor/blocks/links.spec.js#L74-L82

Reply:
Thanks for the suggestion, I was able to find some more useful tips for writing e2e tests at [overusing-snapshots.md](https://github.com/WordPress/gutenberg/blob/21b6a3717c1b09c376f7a5d914b0953aa2cdcddd/docs/contributors/code/e2e/overusing-snapshots.md#L4)

I also notice the `<!-- wp:buttons -->` test across the file and other block tests like `list`, If you have plan to refactor these as well, feel free to count me in when the plan is elaborated